npm/npm.go

```diff
@@ -157,14 +157,22 @@ func (npMgr *NetworkPolicyManager) restore() {
 }
 
 // backup takes snapshots of iptables filter table and saves it periodically.
-func (npMgr *NetworkPolicyManager) backup() {
-	iptMgr := iptm.NewIptablesManager()
-	var err error
-	for {
-		time.Sleep(backupWaitTimeInSeconds * time.Second)
+func (npMgr *NetworkPolicyManager) backup(stopCh <-chan struct{}) {
+	var (
+		err    error
+		iptMgr = iptm.NewIptablesManager()
+		ticker = time.NewTicker(time.Second * time.Duration(backupWaitTimeInSeconds))
+	)
+	defer ticker.Stop()
 
-		if err = iptMgr.Save(util.IptablesConfigFile); err != nil {
-			metrics.SendErrorLogAndMetric(util.NpmID, "Error: failed to back up Azure-NPM states %s", err.Error())
+	for {
+		select {
+		case <-stopCh:
+			return
+		case <-ticker.C:
+			if err = iptMgr.Save(util.IptablesConfigFile); err != nil {
```

Comment:
this err does not need to be function scoped, you can scope it just to this conditional:
```suggestion
			if err := iptMgr.Save(util.IptablesConfigFile); err != nil {
```